### DISCUSSION TITLE
Save every screenshot doesn't matter if just copying or uploading

### DISCUSSION THREAD

**Discussion by CarterSnich** [flameshot-org/flameshot#3206] [Q&A] [answered]
How to auto-save screenshots no matter if just copying or uploading?

**Comment by mmahmoudian** [comment, marked as answer]
Such feature for copying exists, but not for uploads. Please [open a feature request](https://github.com/flameshot-org/flameshot/issues/new?assignees=&labels=Enhancement&projects=&template=feature-request.yaml)

For Copying here is the settings:

![image](https://github.com/flameshot-org/flameshot/assets/390889/c99b49b0-7065-464d-a0c9-dd3d7bd238ac)


Technically it is already possible if you are on macOS or Linux by using CLI. The following will save, upload, and copy to clipboard at the same time:

```sh
flameshot gui --upload --clipboard --path "/home/mehrad/Pictures/"
```

Note that is you have the "Copy file path after save" or "Copy URL after upload", they can over-write the clipboard. In such cases, you can use a clipboard manager (e.g [CopyQ](https://hluk.github.io/CopyQ/)) to retain all that info.
